tests/unit/binlog_async.cc

```diff
@@ -0,0 +1,118 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+
+#include <yatl/lite.h>
+
+#include <libdrizzle-5.1/libdrizzle.h>
+
+#include <cstdio>
+#include <cstdlib>
+
+void binlog_error(drizzle_return_t ret, drizzle_st *connection, void *context)
+{
+  (void) context;
+  ASSERT_EQ_(DRIZZLE_RETURN_EOF, ret, "%s(%s)", drizzle_error(connection), drizzle_strerror(ret));
+}
+
+void binlog_event(drizzle_binlog_event_st *event, void *context)
+{
+  (void) context;
+  uint32_t timestamp;
+  int *cxt_ptr = (int*) context;
+  timestamp= drizzle_binlog_event_timestamp(event);
+
+  ASSERT_EQ_(1, cxt_ptr[0], "Invalid drizzle context: Got '%d', Expected '1'",
+    cxt_ptr[0]);
+
+  /* Test to see if timestamp is greater than 2012-01-01 00:00:00, corrupted
+   * timestamps will have weird values that shoud fail this after several
+   * events.  Also rotate event doesn't have a timestamp so need to add 0
+   * to this test */
+  ASSERT_FALSE_(((timestamp < 1325376000) && (timestamp != 0)),
+    "Bad timestamp retrieved: %u", timestamp);
+
+  /* An event higher than the max known is bad, either we don't know about
+   * new events or type is corrupted */
+  ASSERT_FALSE_((drizzle_binlog_event_type(event) >= DRIZZLE_EVENT_TYPE_END),
+    "Bad event type: %d", drizzle_binlog_event_type(event));
+}
+
+int main(int argc, char *argv[])
+{
+  (void) argc;
+  (void) argv;
+  drizzle_st *con;
+  drizzle_binlog_st *binlog;
+  drizzle_options_st *opts;
+  drizzle_return_t ret;
+
+  opts = drizzle_options_create();
+  drizzle_options_set_non_blocking(opts, true);
+
+  con= drizzle_create(getenv("MYSQL_SERVER"),
+                      getenv("MYSQL_PORT") ? atoi("MYSQL_PORT") : DRIZZLE_DEFAULT_TCP_PORT,
+                      getenv("MYSQL_USER"),
+                      getenv("MYSQL_PASSWORD"),
+                      getenv("MYSQL_SCHEMA"), opts);
+  ASSERT_NOT_NULL_(con, "Drizzle connection object creation error");
+
+  drizzle_set_verbose(con, DRIZZLE_VERBOSE_INFO);
+
+  ret= drizzle_connect(con);
+
+  while (ret != DRIZZLE_RETURN_OK)
+  {
+    drizzle_wait(con);
+    ret = drizzle_connect(con);
+  }
+
+  int cxt = 1;
+
+  binlog= drizzle_binlog_init(con, binlog_event, binlog_error, (void*)&cxt, true);
+  ret= drizzle_binlog_start(binlog, 0, "", 0);
+  SKIP_IF_(ret == DRIZZLE_RETURN_ERROR_CODE, "Binlog is not open?: %s(%s)",
```

Comment:
Does this mean that if there is a problem with the binlog opening it just skips this test? Is that the intended behaviour?